parsers/typespec.c

```diff
@@ -0,0 +1,682 @@
+	{
+		/* This is a property */
+		makeTypeSpecTag(token, K_PROPERTY);
+
+		/* Skip type declaration, modifiers, etc. until ; or , */
+		while (token->type != TOKEN_SEMICOLON &&
+			   token->type != TOKEN_COMMA &&
+			   token->type != TOKEN_CLOSE_CURLY &&
+			   token->type != TOKEN_EOF)
+		{
+		/* Skip generic parameters if any */
+			readToken(token);
+			if (token->type == TOKEN_OPEN_ANGLE)
+			{
+				skipTypeParameters(token);
+			}
+		}
+	}
+}
+
+static void parseOperation(tokenInfo *const token)
+{
+	readToken(token);
+
+	if (token->type == TOKEN_IDENTIFIER)
+	{
+		makeTypeSpecTag(token, K_OPERATION);
+		readToken(token);
+
+		/* Skip generic parameters if any */
+		if (token->type == TOKEN_OPEN_ANGLE)
+		{
+			skipTypeParameters(token);
+		}
+
+		/* Handle "is" relationship */
+		if (token->type == TOKEN_KEYWORD && token->keyword == KEYWORD_is)
+		{
+			/* Skip expression after "is" */
+			do {
+				readToken(token);
+
+				if (token->type == TOKEN_OPEN_ANGLE)
+				{
+					skipTypeParameters(token);
+				}
+
+				if (token->type == TOKEN_SEMICOLON)
+					break;
+
+			} while (token->type != TOKEN_SEMICOLON && token->type != TOKEN_EOF);
+		}
+	}
+}
+
+static void parseInterface(tokenInfo *const token)
+{
+	readToken(token);
+
+	if (token->type == TOKEN_IDENTIFIER)
+	{
+		int ifaceIndex = makeTypeSpecTag(token, K_INTERFACE);
+		readToken(token);
+
+		/* Skip extends clause */
+		if (token->type == TOKEN_KEYWORD && token->keyword == KEYWORD_extends)
+		{
+			while (token->type != TOKEN_OPEN_CURLY && token->type != TOKEN_EOF)
+			{
+				readToken(token);
+			}
+		}
+
+		if (token->type == TOKEN_OPEN_CURLY)
+		{
+			enterScope(token, ifaceIndex);
+		}
+	}
+}
+
+static void parseModel(tokenInfo *const token)
+{
+	readToken(token);
+
+	if (token->type == TOKEN_IDENTIFIER)
+	{
+		int modelIndex = makeTypeSpecTag(token, K_MODEL);
+		readToken(token);
+
+		/* Skip extends clause if any */
+		if (token->type == TOKEN_KEYWORD && token->keyword == KEYWORD_extends)
+		{
+			while (token->type != TOKEN_OPEN_CURLY && token->type != TOKEN_EOF)
+			{
+				readToken(token);
+			}
+		}
+
+		if (token->type == TOKEN_OPEN_CURLY)
+		{
+			enterScope(token, modelIndex);
+		}
+	}
+}
+
+static void parseUnion(tokenInfo *const token)
+{
+	readToken(token);
+
+	if (token->type == TOKEN_IDENTIFIER)
+	{
+		int unionIndex = makeTypeSpecTag(token, K_UNION);
+		readToken(token);
+
+		if (token->type == TOKEN_OPEN_CURLY)
+		{
+			enterScope(token, unionIndex);
+		}
+	}
+}
+
+static void parseAlias(tokenInfo *const token)
+{
+	readToken(token);
+
+	if (token->type == TOKEN_IDENTIFIER)
+	{
+		makeTypeSpecTag(token, K_ALIAS);
+		readToken(token);
+
+		/* Skip generic parameters if any */
+		if (token->type == TOKEN_OPEN_ANGLE)
+		{
+			skipTypeParameters(token);
+		}
+
+		/* Skip the rest until semicolon */
+		skipToSemicolon(token);
+	}
+}
+
+static void enterScope (tokenInfo *const parentToken, int scope)
+{
+	tokenInfo *token = newToken ();
+    tagEntryInfo *e_parent;
```

Comment:
The indentation offset looks wrong. You may use both whitespaces and tabs mixly.